crates/oxc_linter/src/rules/react/no_find_dom_node.rs

```diff
@@ -0,0 +1,158 @@
+use oxc_ast::AstKind;
+use oxc_diagnostics::{
+    miette::{self, Diagnostic},
+    thiserror::Error,
+};
+use oxc_macros::declare_oxc_lint;
+use oxc_span::Span;
+
+use crate::{context::LintContext, rule::Rule, AstNode};
+
+#[derive(Debug, Error, Diagnostic)]
+#[error("Do not use findDOMNode. It doesn’t work with function components and is deprecated in StrictMode. See https://reactjs.org/docs/react-dom.html#finddomnode")]
+#[diagnostic(severity(warning))]
+struct NoFindDomNodeDiagnostic(#[label] pub Span);
+
+#[derive(Debug, Default, Clone)]
+pub struct NoFindDomNode;
+
+declare_oxc_lint!(
+    /// ### What it does
+    /// This rule disallows the use of `findDOMNode`.
+    ///
+    /// ### Why is this bad?
+    /// Facebook will eventually deprecate `findDOMNode` as it blocks certain improvements in React in the future.
+    ///
+    /// ### Example
+    /// ```javascript
+    /// class MyComponent extends Component {
+    ///   componentDidMount() {
+    ///     findDOMNode(this).scrollIntoView();
+    ///   }
+    ///   render() {
+    ///     return <div />;
+    ///   }
+    ///  }
+    /// ```
+    NoFindDomNode,
+    correctness
+);
+
+impl Rule for NoFindDomNode {
+    fn run<'a>(&self, node: &AstNode<'a>, ctx: &LintContext<'a>) {
+        let AstKind::CallExpression(call_expr) = node.kind() else { return };
+        if call_expr.callee.get_identifier_reference().is_some_and(|x| x.name == "findDOMNode") {
+            ctx.diagnostic(NoFindDomNodeDiagnostic(call_expr.span));
+            return;
+        }
+
+        let Some(member) = call_expr.callee.get_member_expr() else { return };
+        let Some(prop_name) = member.static_property_name() else { return };
+        if prop_name == "findDOMNode" {
```

Comment:
Are we missing checking for "React.` here? Seems like eslint-plugin-react missed a test case?

Reply:
I agree checking the name of the callee here would make this rule more robust. Updated in a7bf448f057290468af089f1d779bae249b899a8.

In the latest code, I checked for `React.`, `ReactDOM.`, and `ReactDom.` to cover different import paths and naming conventions.